marlowe-dashboard-client/src/Contract/State.purs

```diff
@@ -1,19 +1,26 @@
-module Contract.State where
+module Contract.State
+  ( defaultState
+  , mkInitialState
+  , handleQuery
+  , handleAction
+  ) where
 
 import Prelude
 import Contract.Types (Action(..), Query(..), Side(..), State, Tab(..), _confirmation, _executionState, _side, _step, _tab)
 import Data.Lens (assign, modifying, use)
 import Data.Maybe (Maybe(..))
 import Data.Unfoldable as Unfoldable
 import Effect.Aff.Class (class MonadAff)
-import Halogen (HalogenM, raise)
-import MainFrame.Types (ChildSlots, Msg(..))
+import Halogen (HalogenM)
+import MainFrame.Types (ChildSlots, Msg)
 import Marlowe.Execution (NamedAction(..), _namedActions, _state, initExecution, merge, mkTx, nextState)
-import Marlowe.Semantics (Contract, Slot, _minSlot)
-import Plutus.PAB.Webserver.Types (StreamToServer(..))
+import Marlowe.Semantics (Contract(..), Slot, _minSlot)
 
-initialState :: Slot -> Contract -> State
-initialState slot contract =
+defaultState :: State
+defaultState = mkInitialState zero Close
```

Comment:
Using `Close` and `zero` as default for when there is no contract is a smell for me (is a bit like using `null` in Java). I think this should be wrapped in some kind of `Maybe` type or custom data type with a `NotStarted` constructor or something like that. It is a bit more work, but I think that will make things much easier in the long run, especially when debugging later.

Reply:
Yeah, we have a general problem here. David and I have been experimenting with different options, and so far have found having a default state (that we know will never be used) is the least awkward. Otherwise you have to handle the `Nothing` cases in the view functions, even though the view functions will never be called in that case. But I agree it is "wrong". Going to leave it for now if you don't mind, and we can revisit it as we go along...